File: backend/debugging/ai.js
require("dotenv").config({ path: "./backend/.env" });
const { GoogleGenerativeAI } = require("@google/generative-ai");

// Initialize with your API key
const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY_2);

async function main() {
  // Get the model
  const model = genAI.getGenerativeModel({ model: "gemini-3.6-flash" });

  // Call generateContentStream
  const result = await model.generateContentStream(
    "Hello! Explain what a Discord bot is in detail."
  );

  // Iterate over the stream as chunks arrive
  for await (const chunk of result.stream) {
    const chunkText = chunk.text();
    process.stdout.write(chunkText);
  }
}

main().catch(console.error);

File: backend/debugging/ai.py
import os
from google import genai

# Read the key from the environment. Never hardcode credentials here:
# this file is committed to the repository, and a literal key published to
# a public remote will be detected and revoked automatically.
#
#   PowerShell:  $env:GEMINI_API_KEY_1 = "your-key"
#   bash:        export GEMINI_API_KEY_1="your-key"
api_key = os.environ.get("GEMINI_API_KEY_1")
if not api_key:
    raise SystemExit(
        "GEMINI_API_KEY_1 is not set. Export it before running this script."
    )

client = genai.Client(api_key=api_key)

stream = client.interactions.create(
    model="gemini-3.6-flash",
    input="Hello! Explain what a Discord bot is.",
    stream=True
)

for event in stream:
    if event.event_type == "step.delta":
        if event.delta.type == "text":
            print(event.delta.text, end="", flush=True)
